llvm/lib/Analysis/VectorUtils.cpp

```diff
@@ -1477,6 +1480,32 @@ void VFABI::getVectorVariantNames(
   }
 }
 
+FunctionType *VFABI::createFunctionType(const VFInfo &Info,
+                                        const FunctionType *ScalarFTy) {
+  ElementCount VF = Info.Shape.VF;
+  // Create vector parameter types
+  SmallVector<Type *, 8> VecTypes;
+  for (auto [STy, VFParam] : zip(ScalarFTy->params(), Info.Shape.Parameters)) {
```

Comment:
This only works because we know any mask parameter will be last?

Given the following code `VecTypes.insert(VecTypes.begin() + OptMaskPos.value(), MaskTy);` allows the mask to be anywhere I think we'd be better served being more explicit. For example:

```
ScalarParamIndex = 0;
foreach (Info.Shape.Parameters) {
  if (ParamKind == GlobalPredicate) {
    VecTypes.push_back(mask_type)
    continue;
  }

  OperandType = getParamType(ScalarParamIndex++)
  if (ParamKind == vector) {
    OperandType = vector_type(OperandType, VF)

  VecTypes.push_back(OperandType)
}
```
Perhaps there's a nicer c++ way to write this but you get the gist?

Reply:
> This only works because we know any mask parameter will be last?

True, because specs guarantees that.
Thanks for the gist, looks better and it'll handle things nicely.